server/backend/feedback.go

```diff
@@ -0,0 +1,142 @@
+package backend
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	pb "github.com/TUM-Dev/Campus-Backend/server/api/tumdev"
+	"github.com/TUM-Dev/Campus-Backend/server/backend/cron"
+	"github.com/TUM-Dev/Campus-Backend/server/model"
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/gofrs/uuid/v5"
+	"github.com/guregu/null"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+// NewFeedback accepts a stream of feedback messages from the client and stores them in the database/file system.
+func (s *CampusServer) NewFeedback(stream pb.Campus_NewFeedbackServer) error {
+	return s.db.WithContext(stream.Context()).Transaction(func(tx *gorm.DB) error {
```

Comment:
nit: Is there a specific reason for putting this entire thing into a transaction? Seemingly this might cause connection exhaustion, if image file uploads take a very long time - just a gut feeling, not a real thing I have seen so far, but it feels a bit wrong. I understand that you create a bunch of entities in between receiving stuff and this is kind of cool to save memory. (Mostly interested in discussing this problem, not that you just go and change stuff here - its fine as is probably)

Reply:
Well, we create feedback and associated files in this request.
I'd like to have both => being more consistent and having less orphaned resources
(I am fully aware, that this is not the cause of the orphaning)

Reply:
Yep, I understand that. I think just to decouple the two:
* First receive all the files and persist them to disk
* Once the entire data stream is received, then start a transaction and insert all the files/feedback
* If the transaction fails, then delete the files

This way you aren't potentially blocking two io operations. Think of one client being super slow (a few kbit/s) and clogging the transaction/connection to the DB with potential write locks.